src/go/rpk/pkg/cli/cluster/partitions/show.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package partitions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/types"
+)
+
+func newShowPartitionAssignmentsCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	type Result struct {
+		Topic     string
+		Partition int
+		Leader    int
+		NodeCore  []string
+	}
+	var (
+		partitions    []int
+		reqPartitions []int
+		results       []Result
```

Comment:
[nit] These 2 variables can be declared inside the Run function and leave here only the variables used in the flags.